contracts/Freeport.sol

```diff
@@ -23,9 +23,19 @@ contract Freeport is /*
     TransferFees,
     */ SimpleExchange {
 
+    /** Name of the collection for open sea.
+     */
+    string public name;
+
     /** Initialize this contract and its dependencies.
      */
     function initialize() public initializer {
         __SimpleExchange_init();
+        name = "DaVinci";
```

Comment:
The initialize function cannot be changed because it has already been called on existing contracts.

The solution is to make a new init function and call it after upgrade (works for both existing and new deployments).

See an axample: https://github.com/Cerebellum-Network/Freeport-Smart-Contracts/blob/c61d72cc400272ae60fe219c1f0e9862475b02b1/migrations/9_upgrade_auction.js#L13

Reply:
Actually, let’s just drop this initialization. It will default to the empty string. The owner will use the function setName.